File: ai-rag-android/app/src/main/java/net/apperdashery/demo/ai_rag_android/store/AppAction.kt
package net.apperdashery.demo.ai_rag_android.store

import net.apperdashery.demo.ai_rag_android.model.OshaStandardResponse

sealed class AppAction {
    data class Osha(val action: OshaAction) : AppAction()
}

sealed class OshaAction {
    data class AskExpertTapped(val query: String) : OshaAction()
    data class ResponseReceived(val response: OshaStandardResponse) : OshaAction()
    data class RequestFailed(val message: String) : OshaAction()
    object DismissedResponse : OshaAction()
    object NavigationHandled : OshaAction()
}

File: ai-rag-app/ai-rag-app/Store/AppAction.swift
//
//  AppAction.swift
//  ai-rag-app
//
//  Created by Graham Diehl on 8/26/26.
//

import Foundation

enum AppAction {
    case osha(OshaAction)
}

enum OshaAction {
    case askExpertTapped(query: String)
    case responseReceived(OshaStandardResponse)
    case requestFailed(String)
    case dismissedResponse
}
